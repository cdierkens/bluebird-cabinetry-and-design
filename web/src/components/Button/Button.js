import { Link } from "gatsby";
import PropTypes from "prop-types";
import React from "react";
import * as styles from "./Button.module.css";

const Button = ({ variant, className, children, to, type, onClick }) => {
  const fullClassName = `${
    variant === "white" ? styles.ButtonWhite : styles.ButtonBlue
  } ${className} focus:outline-none focus:ring`;

  return type === "link" ? (
    <Link to={to} className={fullClassName}>
      {children}
    </Link>
  ) : (
    <button type={type} className={fullClassName} onClick={onClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  variant: PropTypes.oneOf(["white", "blue"]),
  type: PropTypes.oneOf(["link", "button", "submit"]),
  children: PropTypes.node,
  className: PropTypes.string,
  to: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  type: "link",
  variant: "white",
  className: "",
  onClick: () => {},
};

export default Button;
