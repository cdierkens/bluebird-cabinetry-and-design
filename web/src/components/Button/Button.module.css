.Button {
  @apply font-body font-semibold rounded-md border-2 py-2 px-9 text-lg inline-block cursor-pointer;
}

.ButtonWhite {
  @apply Button;
  @apply border-white text-white;
}

.ButtonWhite:hover,
.ButtonWhite:active,
.ButtonWhite:focus,
.ButtonWhiteActive {
  @apply bg-gold text-white;
}

.ButtonBlue {
  @apply Button;
  @apply border-blue-dark text-blue-dark;
}

.ButtonBlue:hover,
.ButtonBlue:active,
.ButtonBlue:focus,
.ButtonBlueActive {
  @apply bg-blue-dark text-white;
}
