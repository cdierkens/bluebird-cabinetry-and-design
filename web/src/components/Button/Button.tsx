import clsx from "clsx";
import { Link } from "gatsby";
import React from "react";
import { tw } from "../../lib/tailwind";
import * as styles from "./Button.module.css";

type ButtonProps = {
  variant?: "white" | "blue";
  className?: string;
} & (
  | {
      type: "link";
      to: string;
      onClick?: never;
    }
  | {
      type?: "button" | "submit";
      to?: never;
      onClick?: React.MouseEventHandler<HTMLButtonElement>;
    }
);

const Button: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  children,
  className,
  onClick,
  to,
  type = "button",
  variant = "white",
}) => {
  const fullClassName = clsx(
    styles.Button,
    {
      [styles.ButtonWhite]: variant === "white",
      [styles.ButtonBlue]: variant === "blue",
    },
    tw`focus:outline-none focus:ring`,
    className
  );

  return type === "link" ? (
    <Link to={to as string} className={fullClassName}>
      {children}
    </Link>
  ) : (
    <button type={type} className={fullClassName} onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
