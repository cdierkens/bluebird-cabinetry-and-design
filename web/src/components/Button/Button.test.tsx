import { render } from "@testing-library/react";
import React from "react";
import Button from "./Button";

it("matches snapshot", () => {
  const { asFragment } = render(<Button />);

  expect(asFragment()).toMatchSnapshot();
});
