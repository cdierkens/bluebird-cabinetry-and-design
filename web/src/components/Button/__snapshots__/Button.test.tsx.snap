// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button ButtonWhite focus:outline-none focus:ring"
    type="button"
  />
</DocumentFragment>
`;
